MapboxAndroidDemo/src/global/java/com/mapbox/mapboxandroiddemo/MainActivity.java

```diff
@@ -251,14 +254,19 @@ public void onItemClicked(RecyclerView recyclerView, int position, View view) {
 
         // in case it's an info tile
         if (model != null) {
-          if (showJavaExamples) {
-            startActivity(model.getJavaActivity());
-          } else {
-            startActivity(model.getKotlinActivity());
-          }
 
-          analytics.clickedOnIndividualExample(getString(model.getTitle()), loggedIn);
-          analytics.viewedScreen(getString(model.getTitle()), loggedIn);
+          // Prevents rapid double click.
+          if (SystemClock.elapsedRealtime() - clickTimeOfLastSelectedExample > CLICK_SPEED_THRESHOLD_MS) {
```

Comment:
This looks to me like a device-dependent way to resolve the issue. How about leveraging `onPause`/`onResume` to remove/add recycler click listener? Is the timing of those lifecycle callbacks correct?